packages/browser-utils/src/metrics/browserMetrics.ts

```diff
@@ -120,6 +121,58 @@ export function startTrackingLongTasks(): void {
   });
 }
 
+/**
+ * Start tracking long animation frames.
+ */
+export function startTrackingLongAnimationFrames(): void {
+  // NOTE: the current web-vitals version (3.5.2) does not support long-animation-frame, so
+  // we directly observe `long-animation-frame` events instead of through the web-vitals
+  // `observe` helper function.
+  if (PerformanceObserver.supportedEntryTypes.includes('long-animation-frame')) {
+    const observer = new PerformanceObserver(list => {
+      for (const entry of list.getEntries() as PerformanceLongAnimationFrameTiming[]) {
+        if (!getActiveSpan()) {
+          return;
+        }
+        if (!entry.scripts[0]) {
+          return;
+        }
+
+        const startTime = msToSec((browserPerformanceTimeOrigin as number) + entry.startTime);
+        const duration = msToSec(entry.duration);
+
+        const attributes: SpanAttributes = { [SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN]: 'auto.ui.browser.metrics' };
+        if (entry.scripts[0]) {
+          attributes['browser.script.invoker'] = entry.scripts[0].invoker;
+          attributes['browser.script.invoker_type'] = entry.scripts[0].invokerType;
+
+          if (entry.scripts[0].sourceURL.length > 0) {
+            attributes['code.filepath'] = entry.scripts[0].sourceURL;
+          }
+          if (entry.scripts[0].sourceFunctionName.length > 0) {
+            attributes['code.function'] = entry.scripts[0].sourceFunctionName;
+          }
+          if (entry.scripts[0].sourceCharPosition !== -1) {
+            attributes['browser.script.source_char_position'] = entry.scripts[0].sourceCharPosition;
+          }
+        }
```

Comment:
We should avoid the `entry.scripts[0]` duplication, as `scripts[0].X` is unminifiable. Assigning to `initialScript` and destructuring helps with this.

```suggestion
        const attributes: SpanAttributes = {
          [SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN]: 'auto.ui.browser.metrics';
        };
        const initialScript = entry.scripts[0];
        if (initialScript) {
          const { invoker, invokerType, sourceURL, sourceFunctionName, sourceCharPosition } = initialScript;
          attributes['browser.script.invoker'] = invoker;
          attributes['browser.script.invoker_type'] = invokerType;
          if (sourceURL) {
            attributes['code.filepath'] = sourceURL;
          }
          if (sourceFunctionName) {
            attributes['code.function'] = sourceFunctionName;
          }
          if (sourceCharPosition !== -1) {
            attributes['browser.script.source_char_position'] = sourceCharPosition;
          }
        }
```

Reply:
makes sense, thanks!